### PR TITLE
WRITING-904 updates to compound indexes per audit

### PR DESCRIPTION
**Pull request by bgrabar**

### CODE REVIEW

File: source/core/index-compound.txt [hunk 1]
```diff
@@ -110,39 +110,41 @@ following:
...
 .. code-block:: javascript
 
    { "item": 1, "location": 1, "stock": 1 }
 
-MongoDB **can** use this index to support queries that include:
+MongoDB can use the index for queries on the following fields, which are
+prefixes of the index:
+
+- ``item``
+- ``item``, ``location``, in that order
```
Comment (kay-kim) [inline source/core/index-compound.txt]: think about what you're saying here with your addition of "in that order"

File: source/core/index-compound.txt [hunk 2]
```diff
@@ -110,39 +110,41 @@ following:
...
-- the ``item`` field,
-- the ``item`` field *and* the ``location`` field,
-- the ``item`` field *and* the ``location`` field *and* the
-  ``stock`` field, or
-- only the ``item`` *and* ``stock`` fields; however, this index
-  would be less efficient than an index on only ``item`` and
-  ``stock``.
+- only on the ``location`` field
+- only on the ``stock`` field
+- only on the ``location`` and ``stock`` fields, no matter the order
```
Comment (kay-kim) [inline source/core/index-compound.txt]: same regarding the "no matter the order"

File: source/core/index-compound.txt [hunk 3]
```diff
@@ -110,39 +110,41 @@ following:
...
 
-MongoDB **cannot** use this index to support queries that include:
+MongoDB **can** use the index to support a query on ``item`` and
+``stock``, even though they are not a prefix. However, the index would not
+be as efficient in supporting the query as would be an index on only
+``item`` and ``stock``.
```
Comment (kay-kim) [inline source/core/index-compound.txt]: This should not be here. Belongs where it used to be.

File: source/core/index-compound.txt [hunk 4]
```diff
@@ -110,39 +110,41 @@ following:
...
 
-- only the ``location`` field,
-- only the ``stock`` field, or
-- only the ``location`` *and* ``stock`` fields.
+If an index is an exact prefix of another index, you usually can remove
+it. For example, if you have an index on ``{ a: 1 }`` and also have a
+compound index on ``{ a: 1, b: 1 }``, and if neither index has a sparse or
+unique constraint, then you can remove the ``{ a: 1 }`` index. MongoDB
```
Comment (kay-kim) [inline source/core/index-compound.txt]: Why put the required case of not being sparse or unique on an example?

**Comment by bgrabar** [comment]
Entered review edits!

File: source/core/index-compound.txt [hunk 5]
```diff
@@ -110,39 +110,41 @@ following:
...
+``{ a: 1 }`` and ``{ a: 1, b: 1 }`` are both prefixes of the index
+``{ a: 1, b: 1, c: 1 }``.
 
-If you have a collection that has a compound index on ``{ a: 1, b:
-1 }``, as well as an index that consists of the prefix of that index,
-i.e. ``{ a: 1 }``, assuming none of the index has a sparse or unique
-constraints, then you can drop the ``{ a: 1 }`` index. MongoDB will be
-able to use the compound index in all of situations that it would have
-used the ``{ a: 1 }`` index.
+If an index is an exact prefix of a second index, and if neither index has
```
Comment (kay-kim) [inline source/core/index-compound.txt]: I think you moved the wrong paragraph.  That's probably my fault -- I probably commented on an outdated commit, which skewed the line. This needs to go down, so organizationally, we want:

Prefixes
----------
def of prefixes of indexes

examples of queries supported by index

examples of queries not supported by index

Can get rid of other indexes that are prefixes.

**Comment by kay-kim** [comment]
merged with edits

**Pull request closed without merging**